src/main/java/org/firmata4j/firmata/device/DeviceConnector.java

```diff
@@ -0,0 +1,63 @@
+package org.firmata4j.firmata.device;
+
+import java.io.IOException;
+import java.util.concurrent.BlockingQueue;
+
+/**
+ * thew hardware than Firmata is implemented on
+ */
+public abstract class DeviceConnector {
```

Comment:
It would be better to have this class as the interface.
I suggest the class' name to be `FirmataTransportInterface`. Then implementation that is common for all the transport interfaces should be in `AbstractFirmataTransport` class and concrete implementations (like `SerialFirmataTransport`) are descendants of the abstract class.